libraries/lib-music-information-retrieval/tests/TatumQuantizationFitBenchmarking.cpp

```diff
@@ -0,0 +1,220 @@
+            truth ?
+               std::make_optional(GetOctaveError(*expected, debugOutput.bpm)) :
+               std::nullopt;
+         sampleValueCsv << (truth ? "true" : "false") << ","
+                        << debugOutput.score << ","
+                        << 60. * debugOutput.tatumQuantization.numDivisions /
+                              debugOutput.audioFileDuration
+                        << "," << debugOutput.bpm << ","
+                        << ToString(debugOutput.timeSignature) << ","
+                        << (error.has_value() ? error->factor : 0.) << ","
+                        << (error.has_value() ? error->remainder : 0.) << ","
+                        << debugOutput.tatumQuantization.lag << "," << wavFile
+                        << "\n";
+         return Sample { truth, debugOutput.score, error };
+      });
+
+   // AUC of ROC curve. Tells how good our loop/not-loop clasifier is.
+   const auto rocInfo = GetRocInfo(
+      samples, loopClassifierSettings.at(tolerance).allowedFalsePositiveRate);
+
+   const auto strictThreshold =
+      GetRocInfo(
+         samples, loopClassifierSettings.at(FalsePositiveTolerance::Strict)
+                     .allowedFalsePositiveRate)
+         .threshold;
+
+   // Get RMS of octave errors. Tells how good the BPM estimation is.
+   const auto octaveErrors = std::accumulate(
+      samples.begin(), samples.end(), std::vector<double> {},
+      [&](std::vector<double> octaveErrors, const Sample& sample) {
+         if (sample.octaveError.has_value())
+            octaveErrors.push_back(sample.octaveError->remainder);
+         return octaveErrors;
+      });
+   const auto octaveErrorStd = std::sqrt(
+      std::accumulate(
+         octaveErrors.begin(), octaveErrors.end(), 0.,
+         [&](double sum, double octaveError) {
+            return sum + octaveError * octaveError;
+         }) /
+      octaveErrors.size());
+
+   std::ofstream summaryFile {
+      std::string(CMAKE_CURRENT_SOURCE_DIR) +
+      "/GetMusicalMeterFromSignalBenchmarkingOutput/summary.txt"
+   };
+
+   summaryFile << std::setprecision(std::numeric_limits<double>::digits10 + 1)
+               << "AUC: " << rocInfo.areaUnderCurve << "\n"
+               << "Strict Threshold (Minutes-and-Seconds): " << strictThreshold
+               << "\n"
+               << "Lenient Threshold (Beats-and-Measures): "
+               << rocInfo.threshold << "\n"
+               << "Octave error RMS: " << octaveErrorStd << "\n"
+               << "Audio file hash: " << hash << "\n";
+
+   constexpr auto previousAuc = 0.9121725731895223;
+
+   // If this changed, then some non-refactoring code change happened. If
+   // `rocInfo.areaUnderCurve > previousAuc`, then there's probably no argument
+   // about the change. On the contrary, though, the change is either an
+   // inadverted bug, and if it is deliberate, should be well justified.
+   REQUIRE(rocInfo.areaUnderCurve == previousAuc);
```

Comment:
From @Paul-Licameli:
> I think you might not get a precise match of numbers if you run the tests an a different platform or CPU

Reply:
That could be a problem. I expect it will be close, though, since I don't expect the false-positive and true-positive rates to differ much or at all. As for the floating-point arithmetic ...
In a follow-up PR I intend to make the evaluation dataset easily available, not only on the machine I'm using right now. When I'm there I'll test it on the macbook I have, see what happens.